app/src/main/java/tech/relaycorp/letro/account/storage/AccountRepository.kt

```diff
@@ -44,20 +57,35 @@ class AccountRepositoryImpl @Inject constructor(
         }
     }
 
-    override suspend fun createAccount(id: String) {
+    override suspend fun createAccount(
+        requestedUserName: String,
+        domainName: String,
+        locale: Locale,
+        veraidPrivateKey: PrivateKey,
+    ) {
         accountDao.insert(
             Account(
-                veraId = id,
+                veraidId = "$requestedUserName@$domainName",
+                requestedUserName = requestedUserName,
+                locale = locale.normaliseString(),
+                veraidPrivateKey = veraidPrivateKey.encoded,
                 isCurrent = true,
             ),
         )
     }
 
-    override suspend fun updateAccountId(id: String, newId: String) {
-        accountDao.getByVeraId(id)?.let {
+    override suspend fun getByRequest(requestedUserName: String, locale: Locale): Account? =
+        accountDao.getByRequestParams(
+            requestedUserName = requestedUserName,
+            locale = locale.normaliseString(),
+        )
+
+    override suspend fun completeRegistration(id: Long, veraidId: String, veraidBundle: ByteArray) {
```

Comment:
Let's pass `Account` here instead of ID, to not make 2 requests to the database, because you've already got the `Account` object on the caller's side. I agree, that maybe pass `id` is better in terms of architecture to make `AccountRepository` responsible for obtain the correct object from the database, but to my mind, it's better to use YAGNI approach here :)

`override suspend fun completeRegistration(account: Account, veraidId: String, veraidBundle: ByteArray)`